docs/src/appendix/cheatcodes/l1_handler_invoke.md

```diff
@@ -0,0 +1,68 @@
+# `l1_handler_invoke`
+
+> `fn invoke(self: L1Handler, from_address: felt252, fee: u128, payload: Span::<felt252>)`
+
+Invokes a `#[l1_handler]` function to mock a
+[message](https://docs.starknet.io/documentation/architecture_and_concepts/L1-L2_Communication/messaging-mechanism/)
+arriving from Ethereum.
+
+```rust
+struct L1Handler {
+    contract_address: ContractAddress,
+    selector_name: felt252,
+}
```

Comment:
I still would prefer the solution I proposed, if we have it this way you need to provide `from_address`, `fee` and `payload` each time even if it is not relevant to the tested code. IMO better interface would be
```
L1HandlerInvoke {
    contract_address: ContractAddress,
    selector_name: felt252,
    from_address: felt252, 
    fee: u128,
    payload: Span::<felt252>
}


trait L1HandlerTrait {
    fn default(contract_address, selector_name);
    fn execute(self: L1Handler);
}
```
and `default` would set all remaining  fields to `0` or sth.

This way interface would be more extensible as you could add more fields without introducing breaking changes. If somebody has to configure some additional parameters then it is easy to just create the structure by hand.

Reply:
Understood, I've made some tests for spelling and usage. The only point I see is the fact that one must not forget to use `mut` on the instance that is returned from the `new` method to customize it (which is mandatory as default empty payload will never fit a use case as it's the cairo deserialization that will fail, not the user code).

Is this ok for you?

1. New was more "array like", it may sound more familiar to users. Or do you definitely prefer `default`?
2. `execute` sounds nice to me too due to the tx that is executed, let's go with it!

```rust
let data: Array<felt252> = array![1, 2];

let mut payload_buffer: Array<felt252> = ArrayTrait::new();
data.serialize(ref payload_buffer);

let mut l1_handler = L1HandlerTrait::new(contract_address, function_name: 'process_l1_message');

// Testing execute here makes almost no sense to me, or I am perhaps missing
// something important. Except perhaps someone sending message without any payload
// and testing the address 0.

l1_handler.from_address = 0x123;
l1_handler.payload = payload.span();

l1_handler.execute();
```

Reply:
SGTM